cli/loader/src/lib.rs

```diff
@@ -929,6 +950,7 @@ fn needs_recompile(
     lib_path: &Path,
     parser_c_path: &Path,
     scanner_path: &Option<PathBuf>,
+    external_files_paths: &[PathBuf],
```

Comment:
May be it's time to generalize this function and accept just a one parameter: `checked_paths: &[Path]`